WooCommerce/Classes/Authentication/Store Creation/Profiler/StoreCreationProfilerQuestionContainerViewModel.swift

```diff
@@ -0,0 +1,113 @@
+import Foundation
+import Yosemite
+
+enum StoreCreationProfilerQuestion: Int {
+    case sellingStatus
+    case category
+    case country
+    case challenges
+    case features
+
+    /// Progress to display for the profiler flow
+    var progress: Double {
+        switch self {
+        case .sellingStatus:
+            return 0.2
+        case .category:
+            return 0.4
+        case .country:
+            return 0.6
+        case .challenges:
+            return 0.8
+        case .features:
+            return 1
+        }
+    }
+
+    var previousQuestion: StoreCreationProfilerQuestion? {
+        .init(rawValue: self.rawValue - 1)
+    }
+}
+
+/// View model for `StoreCreationProfilerQuestionContainer`.
+final class StoreCreationProfilerQuestionContainerViewModel: ObservableObject {
+
+    let storeName: String
+    private let analytics: Analytics
+    private let completionHandler: (SiteProfilerData?) -> Void
+
+    private var storeCategory: StoreCreationCategoryAnswer?
+    private var sellingStatus: StoreCreationSellingStatusAnswer?
+    private var storeCountry: SiteAddress.CountryCode = .US
```

Comment:
The auto detection logic is already handled in the country question. This default value is only there to avoid having to deal with optional - the value will be overridden because the country question is non-optional. Let me know if you think there could be any other edge case here.

Reply:
Oh, I didn't know that. Thanks for the info. All good. 👍